### ISSUE TITLE
AzureResourceGroupDeploymentV2 - Multiple deploys detected

### ISSUE THREAD

**Issue by lindgrend** [microsoft/azure-pipelines-tasks#11814]
## Note  
Issues in this repo are for tracking bugs, feature requests and questions for the tasks in this repo 

For a list:  
https://github.com/Microsoft/azure-pipelines-tasks/tree/master/Tasks

If you have an issue or request for the Azure Pipelines service, use developer community instead:  

https://developercommunity.visualstudio.com/spaces/21/index.html )

## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: AzureResourceGroupDeploymentV2

## Environment
- Server - Azure Pipelines
    
## Issue Description
Hello, we are running a daily time-triggered release to setup our entire build environment on Azure via ARM every morning and then we tear it down again in the evening after business hours. This has been working well until a couple of days ago when we notice that the release is triggering two deploys on Azure. This makes both deploys fail with conflicts. Restarting the release manually seems so far to not have the same problem. 

### Task logs
See screenshot taken from the Azure portal:
![multiple builds started](https://user-images.githubusercontent.com/8226441/69249396-14d5ea80-0bae-11ea-9ec0-205fe56984ec.png)

template-20191120-050047-8561 is the one that was started from Azure devops. template-20191120-050047-9794 is unknown.

## Troubleshooting
I can share all the logs you need in private

### Error logs

I can share all the logs you need in private

**Comment by jalajmsft** [comment]
Can you set the variable "System.debug" with value "true". Redeploy and share logs at RM_Customer_Queries@microsoft.com

**Comment by lindgrend** [comment]
Did you get the logs and are they the ones you expected?

**Comment by jalajmsft** [comment]
Hey @lindgrend sorry for the delayed response, yes we have received the logs. From logs, it seems like the task AzureResourceGroupDeploymentV2 is deploying only once. It could be the case that the release is cloned or some other release is running in parallel to this one when the schedule condition is met.

Can you please check that there is no duplicate of release? Also, can you please share the screenshot of the list of releases/deployments and their history from the Azure DevOps portal.

**Comment by lindgrend** [comment]
I went over all our pipelines again and this time I likely found the badly named culprit, I now disabled the automatic trigger and I'll let you know tomorrow if we can close this (and if that is the case sorry for wasting your time)
